src/Subscriber/MismatchBranchDescriptionSubscriber.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace App\Subscriber;
+
+use App\Api\Issue\IssueApi;
+use App\Event\GitHubEvent;
+use App\GitHubEvents;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\String\UnicodeString;
+
+/**
+ * @author Antoine Makdessi <amakdessi@me.com>
+ */
+class MismatchBranchDescriptionSubscriber implements EventSubscriberInterface
+{
+    private IssueApi $issueApi;
+    private LoggerInterface $logger;
+
+    public function __construct(IssueApi $issueApi, LoggerInterface $logger)
+    {
+        $this->issueApi = $issueApi;
+        $this->logger = $logger;
+    }
+
+    public function onPullRequest(GitHubEvent $event): void
+    {
+        $data = $event->getData();
+        if (!in_array($data['action'], ['opened', 'ready_for_review']) || ($data['pull_request']['draft'] ?? false)) {
+            return;
+        }
+
+        $number = $data['pull_request']['number'];
+
+        $descriptionBranch = $this->extractDescriptionBranchFromBody($data['pull_request']['body']);
+        if (null === $descriptionBranch) {
+            $this->logger->notice('Pull Request without default template.', ['pull_request_number' => $number]);
+
+            return;
+        }
+
+        $targetBranch = $data['pull_request']['base']['ref'];
+        if ($targetBranch === $descriptionBranch) {
+            return;
+        }
+
+        $this->issueApi->commentOnIssue($event->getRepository(), $number, <<<TXT
+Hey!
+
+Thanks for your PR. You are targeting branch "$targetBranch" but it seems your PR description refers to branch "$descriptionBranch".
+Could you update the PR description please or maybe the target branch is wrong? This helps core maintainers a lot.
```

Comment:
This reads a bit weird to me.


```suggestion
Please update the PR description. Or maybe the target branch is wrong? This helps core maintainers a lot.
```

or

```suggestion
Could you please update the PR description or maybe the target branch is wrong? This helps core maintainers a lot.
```

or

```suggestion
Could you update the PR description, please ? Or maybe the target branch is wrong? This helps core maintainers a lot.
```

?

Reply:
Changed to `Could you update the PR description or change target branch? This helps core maintainers a lot.`